packages/core/src/external/aws/dynamodb.ts

```diff
@@ -0,0 +1,117 @@
+import * as AWS from "aws-sdk";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+import { errorToString } from "@metriport/shared";
+import { Config } from "../../util/config";
+import { capture, out } from "../../util";
+
+const region = Config.getAWSRegion();
+
+export type DynamoDbUtilsOptions = {
+  table: string;
+  partitionKey: string;
+  rangeKey?: string;
+  client?: DocumentClient | undefined;
+};
+
+type AttributeValuesMapping = { [k: string]: string | number };
+
+export class DynamoDbUtils {
+  public readonly _docClient: DocumentClient;
+  public readonly _table: string;
+  public readonly _partitionKey: string;
+  public readonly _rangeKey: string | undefined;
+
+  constructor(opts: DynamoDbUtilsOptions) {
+    this._table = opts.table;
+    this._partitionKey = opts.partitionKey;
+    this._rangeKey = opts.rangeKey;
+    this._docClient =
+      opts.client ??
+      new AWS.DynamoDB.DocumentClient({
+        region,
+        apiVersion: "2012-08-10",
+      });
+  }
+
+  /*
+   * Update will insert if not exists
+   */
+  async update({
+    partition,
+    range,
+    expression,
+    expressionAttributesValues,
+    returnValue = "ALL_NEW",
+  }: {
+    partition: string;
+    range?: string;
+    expression: string;
+    expressionAttributesValues?: AttributeValuesMapping;
+    returnValue?: "ALL_OLD" | "ALL_NEW";
+  }): Promise<DocumentClient.UpdateItemOutput> {
+    const { log } = out(`update DDB - table ${this._table} partition ${partition}`);
+    const key = this.createKey(partition, range);
+    const params: DocumentClient.UpdateItemInput = {
+      TableName: this._table,
+      Key: key,
+      UpdateExpression: expression,
+      ...(expressionAttributesValues && { ExpressionAttributeValues: expressionAttributesValues }),
+      ReturnValues: returnValue,
+    };
+    try {
+      return await this._docClient.update(params).promise();
+    } catch (error) {
+      const msg = `Error updating ${this._table} @ DDB`;
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          table: this._table,
+          key,
+          expression,
+          expressionAttributesValues,
+          returnValue,
+          context: "ddb.update",
+          error,
+        },
+      });
```

Comment:
The same chat about where to report errors... I think we should leave that to the higher level code?

Reply:
removed.